src/testing/spec-page.ts

```diff
@@ -282,7 +293,21 @@ function proxyComponentLifeCycles(Cstr: ComponentTestingConstructor) {
   }
 }
 
-function findRootComponent(cmpTags: Set<string>, node: Element): any {
+/**
+ * Return the first Element whose {@link Element#nodeName} property matches a tag found in the provided `cmpTags`
+ * argument.
+ *
+ * This function will inspect each child of the provided `node` argument. If no match is found, the children of the
+ * first child element will be inspected. This processes recurses until either:
```

Comment:
reading the function it looks like it's doing a depth-first search that starts with the first child and then will go through the rest of the children from there - the wording here kind of implies to me that it will only exhaustively search the tree rooted at the first child, rather than exhaustively searching the whole tree